apps/cowswap-frontend/src/utils/iframeResizer.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+export function IframeResizer() {
+  useEffect(() => {
+    // Initial height calculation and message
+    const sendHeightUpdate = () => {
+        const contentHeight = document.body.scrollHeight;
+        window.parent.postMessage({ type: 'iframeHeight', height: contentHeight }, '*');
+    };
+    sendHeightUpdate();
+
+    // Set up a MutationObserver to watch for changes in the DOM
+    const observer = new MutationObserver(() => {
+        // For simplicity, we'll send an update for any mutation, but you can add more specific checks if needed
+        sendHeightUpdate();
```

Comment:
if there's no checks here, `sendHeightUpdate` fn could be used directly in the `MutationObserver` constructor, 
Will we get false positives? Like will we post `iframeHeight` even if it doesn't change?

Reply:
Yes that's right, might run too many times. Should be fixed now in https://github.com/cowprotocol/cowswap/pull/3255